src/TranslationResult.h

```diff
@@ -8,101 +8,178 @@
 public:
-  typedef std::vector<std::pair<std::string_view, std::string_view>>
-      SentenceMappings;
-#ifdef WASM_BINDINGS
-  TranslationResult(const std::string &original, const std::string &translation)
-      : originalText(original), translatedText(translation),
-        sentenceMappings() {}
-#endif
-  TranslationResult(const std::string &original, const std::string &translation,
-                    SentenceMappings &sentenceMappings)
-      : originalText(original), translatedText(translation),
-        sentenceMappings(sentenceMappings) {}
-
-  TranslationResult(TranslationResult &&other)
-      : originalText(std::move(other.originalText)),
-        translatedText(std::move(other.translatedText)),
-        sentenceMappings(std::move(other.sentenceMappings)) {}
-
-#ifdef WASM_BINDINGS
-  TranslationResult(const TranslationResult &other)
-      : originalText(other.originalText),
-        translatedText(other.translatedText),
-        sentenceMappings(other.sentenceMappings) {}
-#endif
-
-  TranslationResult(std::string &&original, std::string &&translation,
-                    SentenceMappings &&sentenceMappings)
-      : originalText(std::move(original)),
-        translatedText(std::move(translation)),
-        sentenceMappings(std::move(sentenceMappings)) {}
-
-#ifndef WASM_BINDINGS
-  TranslationResult &operator=(const TranslationResult &) = delete;
-#else
-  TranslationResult &operator=(const TranslationResult &result) {
-    originalText = result.originalText;
-    translatedText = result.translatedText;
-    sentenceMappings = result.sentenceMappings;
-    return *this;
+  // We can allow empty construction.
+  TranslationResult() {}
+  TranslationResult(AnnotatedBlob &&source, AnnotatedBlob &&translation)
+      : source_(std::move(source)), translation_(std::move(translation)) {
+    assert(source_.numSentences() == translation_.numSentences());
   }
-#endif
 
-  /* Return the original text. */
-  const std::string &getOriginalText() const { return originalText; }
+  void set_scores(std::vector<Quality> &&quality) {
+    scores_ = std::move(quality);
+  }
 
-  /* Return the translated text. */
-  const std::string &getTranslatedText() const { return translatedText; }
+  void set_alignments(std::vector<Alignment> &&alignments) {
+    alignments_ = std::move(alignments);
+  }
+
+  // Access to source as a whole.
+  const std::string &source() const { return source_.blob; };
+
+  // Access to translation as a whole.
+  const std::string &translation() const { return translation_.blob; };
+
+  /// Returns the number of units that are translated. To simplify thinking can
+  /// be thought of as number of sentences (equal in source and target by nature
+  /// of the task).
+  const size_t size() const { return source_.numSentences(); }
 
-  /* Return the Quality scores of the translated text. */
-  const QualityScore &getQualityScore() const { return qualityScore; }
+  const size_t numSourceWords(size_t sentenceIdx) {
```

Comment:
Probably, I didn't understand. What does this function do? Does it return the number of words in a sentence with id `sentenceIdx`.

What does `sentenceIdx` correspond to?

Reply:
I don't think we strictly need this. It's a debugging/prototyping artefact. Will get rid of this.

Reply:
Cool. The same applies to `numTranslationWords` below 👍

Reply:
Attempt to remove this will break the test-app which I'm about to bind to regression-test. Do you have a skeleton of how downstream (extension) will consume the API? If not, I'm inclined towards keeping this in some capacity/rename.

Reply:
Resolved resetting TranslationResult.